category_encoders/one_hot.py

```diff
@@ -193,8 +193,10 @@ def get_dummies(X_in, cols=None):
 
         bin_cols = []
         for col in cols:
-            classes = [x for x in set(X[col].values.tolist()) if np.isfinite(x)]
-            for class_ in classes:
+            col_tuples = [class_map['mapping'] for class_map in ordinal_encoder.mapping if class_map['col'] == col][0]
+            fit_classes = set(col_val[1] for col_val in col_tuples)
+            input_classes = set(x for x in set(X[col].values.tolist()) if np.isfinite(x))
+            for class_ in (fit_classes | input_classes):
```

Comment:
Ah, sorry for the mixup: I meant to say that we should have a column for the -1 class in the case that handle_unknown is true and there is _not_ an unknown value present, so that the same columns are output from transform no matter what.  So if you are imputing unknown classes with -1, you should _always_ see an output column for that class, just like with any other category. Make sense?